### ISSUE TITLE
Feat: git worktree component

### ISSUE THREAD

**Issue by OscarCreator** [nvim-lualine/lualine.nvim#1130]
### Requested feature

Add `worktree` component https://github.com/nvim-lualine/lualine.nvim/blob/master/doc/lualine.txt#L240

### Motivation

When using a `bare` git repo with multiple `worktree`s it would be useful to see which worktree you are currently on.

**Comment by lrnzmonaco** [comment]
It works with the branch component for me.
Can you include some steps to reproduce your issue?

This is what has worked for me:
1.  ``git clone --bare https://github.com/nvim-lualine/lualine.nvim && cd $_``
    Now we have a bare repo of lualine.nvim
2.  ``git add worktree test_work_tree master``
    Create a worktree on branch master
3.  ``nvim README.md``

<img width="1470" alt="Schermata 2023-11-01 alle 17 40 46" src="https://github.com/nvim-lualine/lualine.nvim/assets/147439836/eda48cf0-3441-4cac-8856-2d2cdba091d3">

If this is not what you desire, let me know I'm happy to help! 😀

**Comment by OscarCreator** [comment]
The worktree I mean in this case is `test_work_tree`. So a way to display that folder name is what I am after.

Something like this should the status bar look like:
```
NORMAL | master | test_work_tree | README.md
```

**Comment by lrnzmonaco** [comment]
Something like this?

<img width="1470" alt="Schermata 2023-11-01 alle 22 57 49" src="https://github.com/nvim-lualine/lualine.nvim/assets/147439836/16edaef4-2742-49bb-b6c3-e12a585e7582">

I have maintained the same icon as the branch because I'm unable to find a specific symbol for worktrees.

**Comment by OscarCreator** [comment]
Yes something like that. I have to see if I can find a symbol for it, otherwise it can just be changed (or removed) at a later time.
